### ISSUE TITLE
[core] Support custom extensions

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#929]
Define a mechanism to load, configure, and execute custom extensions.

### Packaging

Extensions may be packaged in 3 ways:
 - single JAR file. Either the extension has no dependencies or it's an uberjar.
 - zip file. Contains a set of JAR files. This includes the extension plus its dependencies.
 - set of JAR files. This includes the extension plus its dependencies.

### Location

Extensions should be placed at `${JRELEASER_HOME}/extensions/<extension-name>/`. However, this location may be overridden by configuring an explicit value via the DSL

### DSL

Here's how it may look like

```yaml
extensions:
  foo:
    providers:
      - type: com.github.foo.MustacheFunctionProvider
        properties:
          maxLength: 20
  bar:
    directory: /path/to/the/jars
  qux:
    enabled: false
```

Extension `foo` will be loaded from `${JRELEASER_HOME}/extensions/foo`. It may be packaged as `foo-x.y.z.jar`, `foo.-x.y.z.zip`, or as a set of JAR files inside that directory.
It lists an explicit type `com.github.foo.MustacheFunctionProvider` which requires additional settings upon initialization. These settings are defined using the `properties` map. There may be more than one provider type hence a list. 

Extension `bar` will be loaded from `/path/to/the/jars`. It may be packaged as `bar-x.y.z.jar`, `bar.-x.y.z.zip`, or as a set of JAR files inside that directory.

### Loading Mechanism

Each extension will be loaded and initialized upon bootstrap after resolving the model. Extensions will be treated as singletons per each invocation session. JReleaser will use the `ServiceLoader` SPI interface to locate, load, and initialize extensions.

**Comment by aalmiray** [comment]
Also consider an option for declaring GAV coordinates to download an extension and its transitive dependencies.

**Comment by aalmiray** [comment]
Released in v1.3.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.3.0
